components/Form/index.js

```diff
@@ -66,9 +72,24 @@ export default function Form({ onSubmit }) {
       formObject.image = image;
     }
 
-    localStorage.setItem("favoriteLocation", JSON.stringify(formObject));
+    formObject.id = Date.now().toString();
+
+    const existingLocations =
+      JSON.parse(localStorage.getItem("favoriteLocations")) || []; // Retrieve existing favourite locations or build an empty array
+
+    existingLocations.push(formObject); // Add new favourite location to the existing array
 
-    onSubmit(formObject);
+    localStorage.setItem(
+      "favoriteLocations",
+      JSON.stringify(existingLocations)
+    ); // Save the updated array in local storage
```

Comment:
This works. However, there are potential errors that can happen. If you want to protect yourself against them, use the `useLocalStorageState` hook that comes with the package that you'll need to install first, of course.